장병준 1주차 풀이 업로드

ByeongJun-Jang opened the pull request:
주석 처리한 부분은 런타임 에러가 발생해 처리했습니다.

런타임 에러가 발생하는 이유가 배열 크기에 맞지 않으면 발생하는 것으로 알고 있는데,
입출력을 하고 작성된 배열을 나열하면 알맞게 들어간 것은 확인했습니다.
하지만 백준 컴파일러 상에서 에러가 발생했는데 이유가 있을까요..?

ByeongJun-Jang committed:
feat : 장병준 방배정 풀이 업로드

Hong-sungshin commented:
올려주신 코드 그대로 백준에 제출했을때 컴파일 에러가 발생하신 걸까요 ?? 올려주신 것 그대로 제출하면 에러 없이 100점 뜨긴 하는데 아니면, 주석처리한 부분까지 포함시켜 제출했을 때 오류없이 처리하고싶으신것일까요 ??

Berygna commented:
주석을 포함해서 올려주신 코드로 백준에 올려봐도 오류나지 않습니다. 에러날게 없어보이는데 이상하네요....

tjrmswo reviewed (approved):
P5: 좋아요 잘하신 것 같습니다!

ByeongJun-Jang commented:
> 올려주신 코드 그대로 백준에 제출했을때 컴파일 에러가 발생하신 걸까요 ?? 올려주신 것 그대로 제출하면 에러 없이 100점 뜨긴 하는데 아니면, 주석처리한 부분까지 포함시켜 제출했을 때 오류없이 처리하고싶으신것일까요 ??

아 제가 설명이 부족했던거 같습니다. 주석부분으로 데이터를 받을 때 런타임에러가 발생하여 과정을 진행시키지 않고 따로 반복문으로 학생 배열을 입력했습니다. 
런타임 에러가 발생한 부분은 데이터 함수이고 이 부분이 해결되지 않아 반복문으로 데이터 입력을 받았다고 생각하시면 될 거 같아요 
설명이 부족해 죄송합니다 ㅠㅠ

ByeongJun-Jang commented:
> 주석을 포함해서 올려주신 코드로 백준에 올려봐도 오류나지 않습니다. 에러날게 없어보이는데 이상하네요....

제가 설명이 부족했습니다 ㅠㅠ
런타임 에러가 발생한 부분은 데이터 함수이고 이 부분이 해결되지 않아 반복문으로 데이터 입력을 받았다고 생각하시면 될 거 같아요
그래서 질문이였던 부분은 재귀함수로 데이터를 받으면 런타임 에러가 발생하여 작성했습니다

ByeongJun-Jang commented:
자바로 재귀함수로 받아오면 100점으로 나오네요 ㅠㅠ
파이썬에서 데이터란 함수로 받으면 12점 런타임에러가 발생하고 그 안에 데이터 입력을 함수 안에서가 아닌 밖에서 반복문으로 받으면 100점이 나와 너무 넌센스처럼 느껴져서 질문 올린거 였습니당.. 
이 점은 차후에 찾아보도록 하겠습니다
```
// 자바코드
import java.util.Scanner;

public class test {
    public static Scanner scanner = new Scanner(System.in); 

    public static void main(String[] args) {
        int n = scanner.nextInt(); 
        int k = scanner.nextInt(); 
        int[][] student = new int[2][6]; 

        student = 입력받기(0, n, student); 

        int count = 방개수계산(student, k); 
        System.out.println(count);
    }

    public static int[][] 입력받기(int num, int n, int[][] student) {
        if (num == n) {
            return student;
        }

        int s = scanner.nextInt(); 
        int y = scanner.nextInt(); 
        student[s][y - 1]++; 

        return 입력받기(num + 1, n, student);
    }

    public static int 방개수계산(int[][] student, int k) {
        int count = 0;
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < 6; j++) {
                if (student[i][j] % k == 0) {
                    count += student[i][j] / k; 
                } else {
                    count += student[i][j] / k + 1; 
                }
            }
        }
        return count;
    }
}
```
```
# 파이썬코드
n,k=map(int,input().split())
student = [[0] * 6 for _ in range(2)] 
def 데이터(num, student):
    if num == n:
        return student

    s, y = map(int, input().split())
    student[s][y-1] += 1

    return 데이터(num+1, student)

student = 데이터(0, student)

count = 0
for i in range(2):
    for j in range(6):
        if(student[i][j]%k == 0):
            count += student[i][j]/k
        else:
            count += student[i][j]//k + 1
    
print(int(count))
```

SWARVY commented on BOJ\_13300.py:
```diff
@@ -0,0 +1,27 @@
+        else:
+            count += student[i][j]//k + 1
+    
+print(int(count))
```
p1: EOL 처리를 해주시는게 좋을 것 같습니다!
[EOL](https://github.com/2024-goormKDT-goormBread/Algorithm_Study/pull/1#discussion_r1479823282)

혜민님께 드렸던 리뷰로 설명을 대신할게용

SWARVY commented on BOJ\_13300.py:
```diff
@@ -0,0 +1,27 @@
+    student[s][y-1] += 1
+
+count = 0
+for i in range(2):
```
p5: 기능별로 함수로 나누는 연습을 하시면 나중에 로직 짜기가 편해집니다!

ByeongJun-Jang replied:
넵 ! 두 의견 확인했습니다 ! 감사합니다 !

SWARVY reviewed (commented):
수고하셨어요 :)
